bot/plugins/instance.py

```diff
@@ -395,41 +469,69 @@ async def execute(self) -> None:
             await plugin.app.rest.trigger_typing(self.channel)
 
         # try to execute code
-        att: hikari.Bytes | None = None
-        out: str | None
+        code_attr: hikari.Bytes | None = None
+        stdin_attr: hikari.Bytes | None = None
+        out: list[str] = [f"<@{self.requester}>"]
+
+        code_output_in_file = False
+        stdin_in_file = False
+
+        code_output: str = ""
         if self.runtime:
             ret = await self.runtime.provider.execute(self)
-            out = ret.format()
+            code_output = ret.format()
 
-            if len(out) > 1_950:
-                att = hikari.Bytes(out, "output.ansi")
-                out = None
-            else:
-                if out in {"", "\n"}:
-                    out = "Your code ran with no output."
-                else:
-                    out = f"```ansi\n{out}\n```"
+            if not code_output.strip():
+                out += ["Your code ran with no output."]
+                code_output = ""
         else:
-            out = "No runtime selected."
+            out += ["No runtime selected."]
 
-        if out:
-            out = f"<@{self.requester}>\n{out}"
-        else:
-            out = f"<@{self.requester}>"
+        stdin: str = self.stdin or ""
+
+        if len(code_output) + len(stdin) > 1_950:
+            if len(code_output) < 1_950:
+                stdin_in_file = True
+            elif len(stdin) < 1_950:
+                code_output_in_file = True
+            else:
+                stdin_in_file = True
+                code_output_in_file = True
+
+        if code_output:
+            if code_output_in_file:
+                code_attr = hikari.Bytes(code_output, "code.ansi")
+            else:
+                out += [f"```ansi\n{code_output}```"]
+
+        if stdin:
+            if stdin_in_file:
+                stdin_attr = hikari.Bytes(stdin, "stdin.txt")
+            else:
+                stdin = "\n".join(
+                    f"\x1b[1;33mIN:\x1b[0m {line}" for line in stdin.splitlines()
+                )
```

Comment:
This at the very least needs to happen before the len() check. An input could be 1800 newlines, which would be under 1950, but far over with all the formatting this adds.

I'm not sure if it's a good idea to also have this formatting in the attachments, but it's worth noting that if the file extension is .ansi it will support ansi the same way codeblocks do.

Reply:
I don't think its a good idea to put the formatting in the file. Its in the message to add clarity, I don't think it adds clarity when the file is already named `stdin.txt`.

Reply:
Fair enough. Still need to handle the len() thing though